docker-credential-ecr-login can only be used with Amazon Elastic Container Registry

### Summary 
Build fails while attempting to pull images from Docker Hub.
### Steps to reproduce the behavior
Run any build with a Dockerfile that has a base image from Docker Hub (or anywhere besides ECR) in an EKS 1.13 cluster with ECR enabled for classic jenkins.

### Expected behavior
The docker build will pull the base image from Docker Hub without error when running the `skaffold build` command.
### Actual behavior
Build fails when building the docker images with skaffold with an error stating docker-credential-ecr-login can only be used with ECR.
```
Building [stream-worker]...
time="2019-06-28T15:23:11Z" level=warning msg="Error processing base image (python:3.6) for ONBUILD triggers: getting remote config: getting image: Get https://auth.docker.io/token?scope=repository%3Alibrary%2Fpython%3Apull&service=registry.docker.io: invoking docker-credential-ecr-login: exit status 1; output: 2019-06-28T15:23:11Z [ERROR] Error parsing the serverURL: https://index.docker.io, error: docker-credential-ecr-login can only be used with Amazon Elastic Container Registry.\ncredentials not found in native keychain. Dependencies may be incomplete."
Sending build context to Docker daemon 
```

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 2.0.71
Kubernetes cluster v1.13.7-eks-c57ff8
kubectl            v1.15.0
helm client        v2.10.0+g9ad53aa
helm server        v2.10.0+g9ad53aa
git                git version 2.17.1
Operating System   Ubuntu 18.04.2 LTS
```

### Jenkins type

<!--
Select which installation type are you using. 
-->
- [ ] Next Generation (Tekton + Prow)
- [X] Classic Jenkins
- [ ] Serverless Jenkins (JenkinsFileRunner + Prow)

### Kubernetes cluster

EKS 1.13 created with eksctl and ECR.

### Operating system / Environment

Ubuntu 18.04.2 LTS

I also updated my k8s secret, with content:

```
{
  "credsStore": "ecr-login",
  "credHelpers": {
        "ecr.us-west-2.amazonaws.com": "ecr-login"
  }
}
```
Even restarted Jenkins, but it did not help and I'm still getting error:
```
ONBUILD instructions: parsing ONBUILD instructions: processing base image (composer:1) for ONBUILD triggers: getting remote config: getting image: Get https://auth.docker.io/token?scope=repository%3Alibrary%2Fcomposer%3Apull&service=registry.docker.io: invoking docker-credential-ecr-login: exit status 1; output: 2019-09-10T12:40:53Z [ERROR] Error parsing the serverURL: https://index.docker.io, error: docker-credential-ecr-login can only be used with Amazon Elastic Container Registry.\ncredentials not found in native keychain"
```

It's worked! partially ..

Docker pull is now working for only 1 of 2 images. The one that works manages to be built and pushed to ECR. Second one fails.

### build

```
 - company-api-nginx: Not found. Building
 - company-api-php-fpm: time="2019-09-23T20:35:36Z" level=warning msg="error checking cache, caching may not work as expected: getting hash for artifact company-api-php-fpm: getting dependencies for company-api-php-fpm: expanding ONBUILD instructions: parsing ONBUILD instructions: processing base image (php:7-fpm-alpine) for ONBUILD triggers: getting remote config: getting image: Get https://auth.docker.io/token?scope=repository%3Alibrary%2Fphp%3Apull&service=registry.docker.io: invoking docker-credential-ecr-login: exit status 1; output: 2019-09-23T20:35:35Z [ERROR] Error parsing the serverURL: https://index.docker.io, error: docker-credential-ecr-login can only be used with Amazon Elastic Container Registry.\ncredentials not found in native keychain"
```

### push for nginx

```
Successfully tagged xxxxxxx.dkr.ecr.eu-west-2.amazonaws.com/company/company-api-nginx:0.0.0-SNAPSHOT-PR-628-3
The push refers to repository [xxxxxxxx.dkr.ecr.eu-west-2.amazonaws.com/company/company-api-nginx]
e929d9605539: Preparing
72c96a498309: Preparing
a0a699786342: Preparing
8b5b6870672a: Preparing
03fb8dd8a4de: Preparing
0665fde51bac: Preparing
3e76d2df1790: Preparing
03901b4a2ea8: Preparing
0665fde51bac: Waiting
3e76d2df1790: Waiting
03901b4a2ea8: Waiting
72c96a498309: Pushed
e929d9605539: Pushed
a0a699786342: Pushed
3e76d2df1790: Layer already exists
03901b4a2ea8: Layer already exists
03fb8dd8a4de: Pushed
0665fde51bac: Pushed
8b5b6870672a: Pushed
0.0.0-SNAPSHOT-PR-628-3: digest: sha256:00a6f1813538ceb9e6296c8741d987e48d010b4b5940b05e80c96b53ceb2a36c size: 1986
```

### no push for php-fpm as it fails at build stage

```
Building [company-api-php-fpm]...
Sending build context to Docker daemon 

time="2019-09-23T20:35:57Z" level=fatal msg="build failed: build failed: building [company-api-php-fpm]: build artifact: docker build: error during connect: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.38/build?buildargs=null&cachefrom=null&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=docker%2Fphp-fpm%2FDockerfile&labels=null&memory=0&memswap=0&networkmode=&rm=0&shmsize=0&t=249877450024.dkr.ecr.eu-west-2.amazonaws.com%2Fcompany%2Fcompany-api-php-fpm%3A0.0.0-SNAPSHOT-PR-628-3&target=&ulimits=null&version=: creating docker context: getting relative tar paths: expanding ONBUILD instructions: parsing ONBUILD instructions: processing base image (php:7-fpm-alpine) for ONBUILD triggers: getting remote config: getting image: Get https://auth.docker.io/token?scope=repository%3Alibrary%2Fphp%3Apull&service=registry.docker.io: invoking docker-credential-ecr-login: exit status 1; output: 2019-09-23T20:35:57Z [ERROR] Error parsing the serverURL: https://index.docker.io, error: docker-credential-ecr-login can only be used with Amazon Elastic Container Registry.\ncredentials not found in native keychain"
```

--- a/kubeProviders/eks/values.tmpl.yaml
+++ b/kubeProviders/eks/values.tmpl.yaml
@@ -5,7 +5,13 @@ jenkins-x-platform:
     # lets enable ECR docker builds
     DockerConfig: |-
       {
+{{- if .Requirements.cluster.registry }}
+        "credHelpers": {
+          "{{ .Requirements.cluster.registry }}": "ecr-login"
+        }
+{{- else }}
         "credsStore": "ecr-login"
+{{- end }}
       }
       
 docker-registry:

